include/pybind11/pytypes.h

```diff
@@ -224,6 +224,18 @@ inline handle get_function(handle value) {
     return value;
 }
 
+// Helper aliases/functions to support implicit casting of values given to python accessors/methods.
+// When given a pyobject, this simply returns the pyobject as-is; for other C++ type, the value goes
+// through pybind11::cast(obj) to convert it to an `object`.
+template <typename T, enable_if_t<is_pyobject<T>::value, int> = 0>
+T object_or_cast(T o) { return o; }
```

Comment:
Would it make sense to accept an universal reference as T here? In that case, I think that values could be moved into the ``object_or_cast`` parameter, avoiding a useless incref/decref.